.github/workflows/sample-build.yml

```diff
@@ -0,0 +1,130 @@
+    ${{ github.workspace }}/node_modules
+  CACHED_BUILD_PATHS: |
+    ${{ github.workspace }}/dist
+  BUILD_CACHE_KEY: ${{ inputs.build_cache_key }}
+
+jobs:
+  job_sample_build:
+    name: Sample Build (${{ matrix.bump }})
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        include:
+          - bump: 'v6'
+            SAMPLE_DIR: 'example/ionic-angular-v6'
+            JAVA_VERSION: '17'
+          - bump: 'v7'
+            SAMPLE_DIR: 'example/ionic-angular-v7'
+            JAVA_VERSION: '21'
+          - bump: 'sample-vue'
+            SAMPLE_DIR: 'example/ionic-vue3'
+            JAVA_VERSION: '17'
+    steps:
+      - name: Check out current commit (${{ github.sha }})
+        uses: actions/checkout@v4
+      - run: corepack enable
+      - name: Set up Node
+        uses: actions/setup-node@v5
+      - name: Check dependency cache
+        uses: actions/cache@0400d5f644dc74513175e3cd8d07132dd4860809
+        with:
+          path: ${{ env.CACHED_DEPENDENCY_PATHS }}
+          key: ${{ inputs.dependency_cache_key }}
+      - name: Check build cache
+        uses: actions/cache@0400d5f644dc74513175e3cd8d07132dd4860809
+        with:
+          path: ${{ env.CACHED_BUILD_PATHS }}
+          key: ${{ env.BUILD_CACHE_KEY }}
+      - name: Install yalc
+        run: yarn  global add yalc
+      - name: Install Ionic
+        run: yarn  global add @ionic/cli
+
+      - name: Download Sentry Capacitor package
+        uses: actions/download-artifact@v4
+        with:
+          name: yalc-package
+          path: ~/.yalc/packages/@sentry/capacitor/
+
+      - name: Publish test broken node.
+        working-directory: example/broken_node_module
+        run: |
+          yarn install
+          yarn build
+          yalc publish
+
+      - name: Build Sample ${{ matrix.bump }}
+        run: yarn run bump:${{ matrix.bump }}
+
+      - name: Set up Java ${{ matrix.JAVA_VERSION }}
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: ${{ matrix.JAVA_VERSION }}
+
+      - name: Make gradlew executable
+        working-directory: ${{ matrix.SAMPLE_DIR }}/android
+        run: chmod +x ./gradlew
+
+      - name: Configure Android build compatibility
+        working-directory: ${{ matrix.SAMPLE_DIR }}/android
+        run: |
+          echo "android.useFullClasspathForDexingTransform=true" >> gradle.properties
+          echo "android.enableR8.fullMode=false" >> gradle.properties
+
+      - name: Android build (Gradle)
+        working-directory: ${{ matrix.SAMPLE_DIR }}/android
+        env:
+          GRADLE_OPTS: -Dorg.gradle.daemon=false
+        run: ./gradlew --no-daemon assembleDebug
+
+      - name: Upload Android APK
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.bump }}-android-apk
+          if-no-files-found: warn
+          path: |
+            ${{ matrix.SAMPLE_DIR }}/android/app/build/outputs/apk/debug/*.apk
```

Comment:
q: Should we upload the logs for Android too instead of the apk?

Reply:
for the time being, we can rely on the `gradlew` log from the action.